framework/scripts/generate_test_list.py

```diff
@@ -72,26 +72,47 @@ def main():
             test_list.append(l)
             all_tests[l] = 1
 
+    has_any_lists = len(test_lists) > 0
+
     # generate the .cpp file
     with open(cpp_file, 'w') as f:
         f.write('#include <span>\n')
-        f.write(f'#include "{h_file_basename}"\n\n')
-        f.write('using namespace std;\n\n')
+        f.write(f'#include "{h_file_basename}"\n')
+        f.write('#include "sandstone_tests.h"\n')
+        f.write('#include "SelectorFactory.h"\n')
         decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
         f.write('\n'.join(decls))
         f.write('\n\n')
+
         for name, tests in test_lists.items():
-            f.write(f'static constexpr struct test * const {name}_test_list[] = {{')
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
             test_list = [ f'    &_test_{test}' for test in tests ]
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
+            f.write(f'static const struct TestList {name}_test_list = {{ {"cpu_" + name if name != "default" else 0}, "{name}", {name}_test_list_ }};\n\n')
+
+        if has_any_lists:
+            f.write('static const struct TestList* test_lists[] = {\n')
+            for name, _ in test_lists.items():
+                if default_name is not None and name != default_name:
+                    f.write(f'    &{name}_test_list,\n')
+            if default_name is not None:
+                f.write(f'    &{default_name}_test_list\n') # put default list as last one
+            f.write('};\n\n')
```

Comment:
This is creating an array of pointers:
```c++
static const struct TestList* test_lists[] = {
    &icx_test_list,
    &cpx_test_list,
    &skx_test_list,
    &spr_test_list,
    &default_test_list
};
```

So there's one indirection too many: we could have a simple array of `TestList` items.

Reply:
This list is in incorrect order (the order from the option had been agreed, selector factory will use default list in most cases).